### DISCUSSION TITLE
Menu's dropdown on Chart is being offset

### DISCUSSION THREAD

**Discussion by somguyth** [reactivemarkets/react-financial-charts#554] [Q&A] [answered]
I'm adding a `chakra-ui` menu dropdown button to a `ChartCanvas`. 

For some reason, when I click on the menu, there will be whitespace between the button and the dropdown menu. This only happens when I put the menu onto the chart. If I have the menu standalone in the browser, it'll work as expected.

[![enter image description here][1]][1]

This is the menu code:

```
function TestMenu() {
  return (
    <g className="react-financial-charts-enable-interaction">
      <foreignObject
        x={30}
        y={30}
        width={"100%"}
        height={"100%"}
        style={{ overflow: "auto" }}
      >
        <Menu>
          <MenuButton as={Button} rightIcon={<ChevronDownIcon />}>
            Actions
          </MenuButton>
          <MenuList>
            <MenuItem>Download</MenuItem>
            <MenuItem>Create a Copy</MenuItem>
            <MenuItem>Mark as Draft</MenuItem>
            <MenuItem>Delete</MenuItem>
            <MenuItem>Attend a Workshop</MenuItem>
          </MenuList>
        </Menu>
      </foreignObject>
    </g>
  );
}
```

This is the full codesandbox:

https://codesandbox.io/s/nervous-haze-3mz2c?file=/src/BasicLineSeries.tsx:511-1199

Any ideas?

  [1]: https://i.stack.imgur.com/uJRQJ.png

**Comment by markmcdowell** [comment]
I can see the menu is adding a transform that's moving the menu. I'm not sure how it's calculating that though, bit difficult to say.

**Comment by somguyth** [comment, marked as answer]
I had to wrap `MenuList` inside a `Portal` and set `zIndex={10}` inside the `MenuList` props.
